src/libraries/Microsoft.Extensions.Http/src/DependencyInjection/SocketsHttpHandlerBuilderExtensions.cs

```diff
@@ -0,0 +1,225 @@
+    /// </summary>
+    public static class SocketsHttpHandlerBuilderExtensions
+    {
+        /// <summary>
+        /// Adds a delegate that will be used to configure the primary <see cref="SocketsHttpHandler"/> for a
+        /// named <see cref="HttpClient"/>.
+        /// </summary>
+        /// <param name="builder">The <see cref="ISocketsHttpHandlerBuilder"/>.</param>
+        /// <param name="configure">A delegate that is used to modify a <see cref="SocketsHttpHandler"/>.</param>
+        /// <returns>An <see cref="ISocketsHttpHandlerBuilder"/> that can be used to configure the handler.</returns>
+        [UnsupportedOSPlatform("browser")]
+        public static ISocketsHttpHandlerBuilder Configure(this ISocketsHttpHandlerBuilder builder, Action<SocketsHttpHandler, IServiceProvider> configure)
+        {
+            builder.Services.Configure<HttpClientFactoryOptions>(builder.Name, options =>
+            {
+                options.HttpMessageHandlerBuilderActions.Add(b =>
+                {
+                    if (b.PrimaryHandler is not SocketsHttpHandler socketsHttpHandler)
+                    {
+                        string message = SR.Format(SR.SocketsHttpHandlerBuilder_PrimaryHandlerIsInvalid, nameof(b.PrimaryHandler), typeof(SocketsHttpHandler).FullName, Environment.NewLine, b.PrimaryHandler?.ToString() ?? "(null)");
+                        throw new InvalidOperationException(message);
+                    }
+
+                    configure(socketsHttpHandler, b.Services);
+                });
+            });
+            return builder;
+        }
+
+        /// <summary>
+        /// Uses <see cref="IConfiguration"/> to configure the primary <see cref="SocketsHttpHandler"/> for a
+        /// named <see cref="HttpClient"/>.
+        /// </summary>
+        /// <param name="builder">The <see cref="ISocketsHttpHandlerBuilder"/>.</param>
+        /// <param name="configuration">Configuration containing properties of <see cref="SocketsHttpHandler"/>.</param>
+        /// <returns>An <see cref="ISocketsHttpHandlerBuilder"/> that can be used to configure the handler.</returns>
+        /// <remarks>
+        /// <para>
+        /// Only simple (of type `bool`, `int`, <see cref="Enum"/> or <see cref="TimeSpan"/>) properties of <see cref="SocketsHttpHandler"/> will be parsed.
+        /// </para>
+        /// <para>
+        /// All unmatched properties in <see cref="IConfiguration"/> will be ignored.
+        /// </para>
+        /// </remarks>
+        [UnsupportedOSPlatform("browser")]
+        public static ISocketsHttpHandlerBuilder Configure(this ISocketsHttpHandlerBuilder builder, IConfiguration configuration)
+        {
+            SocketsHttpHandlerConfiguration parsedConfig = ParseSocketsHttpHandlerConfiguration(configuration);
+            return Configure(builder, (handler, _) => FillFromConfig(handler, parsedConfig));
+        }
+
+        [UnsupportedOSPlatform("browser")]
```

Comment:
Unless I'm missing something, we don't need `[UnsupportedOSPlatform("browser")]` on internals, since it's not visible for the related analyzer.
```suggestion
```

Reply:
Apparently, it is still visible, and analyzer will complain that I use `SocketsHttpHandler` inside a method that is reachable on all platforms.